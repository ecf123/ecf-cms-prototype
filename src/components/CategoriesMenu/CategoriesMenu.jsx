import React from 'react'
import './CategoriesMenu.scss'

const CategoriesMenu = ({links}) => {

    typeof links === "string" ? (links = [links]) : links;
      
    const linksArr = links?.map((link, index) => {
        (true ?? ) : ()
        if (index % 2 == 0) {
        return (
            <li key={index} className="categories-menu__item categories-menu__item--purple">{link}</li>);
            } else {
            return (
              <li key={index} className="categories-menu__item categories-menu__item--darkpurple">{link}</li>
            );
        }
    });

    return (
        <ul className="categories-menu">{linksArr}</ul>
    )
}

export default CategoriesMenu