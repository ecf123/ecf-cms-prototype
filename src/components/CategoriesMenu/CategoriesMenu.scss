@use "../../styles/variables" as *;

.categories-menu {
  list-style: none;
  display: flex;
  flex-wrap: wrap;

  &__item {
    font-weight: $font-weight-regular;
    font-size: $font-size-extra-small;
    line-height: 1.2rem;
    margin: 0 0.6rem;
    border-radius: 0.6rem;
    padding: 0 0.6rem;

    &--marketplace {
      color: $color-grey-dark;
      padding-right: 2rem;
      font-size: 1.25rem;
    }

    &--icons {
      padding-right: 0.75rem;
    }

    &--purple {
      color: $color-purple;
      background-color: #f9f5ff;
    }

    &--darkpurple {
      color: $color-purple-dark;
      background-color: #f8f9fc;
    }
  }
}
