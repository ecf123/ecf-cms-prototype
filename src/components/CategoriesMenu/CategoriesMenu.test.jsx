import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";

import CategoriesMenu from "./CategoriesMenu.jsx";

const links = ["one", "two", "three"];

it("should render the category options", () => {
  //Arrange
  render(<CategoriesMenu links={links} condition={false} />);
  //Act
  const link = screen.getByText("one");
  //Assert
  expect(link).toBeInTheDocument();
});

it("should render icon images if condition set to true", () => {
  //Arrange
  render(<CategoriesMenu links={links} condition={true} />);
  //Act
  const icons = screen.getAllByAltText("Icon");
  //Assert
  icons.forEach((icon) => {
    expect(icon).toBeInTheDocument();
  });
});

it("Should render the category options with props", () => {
  const { container } = customRender(
    <CategoriesMenu links={links} condition={false} />
  );
  expect(container).toMatchSnapshot();
});
