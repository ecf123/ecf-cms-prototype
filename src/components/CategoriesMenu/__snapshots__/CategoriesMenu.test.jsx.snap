// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the category options with props 1`] = `
<div>
  <ul
    class="categories-menu"
  >
    <li
      class="categories-menu__item categories-menu__item--purple"
    >
      one
    </li>
    <li
      class="categories-menu__item categories-menu__item--darkpurple"
    >
      two
    </li>
    <li
      class="categories-menu__item categories-menu__item--purple"
    >
      three
    </li>
  </ul>
</div>
`;
